java-manta-client-unshaded/src/main/java/com/joyent/manta/client/MantaClient.java

```diff
@@ -1144,6 +1197,7 @@ public MantaObjectResponse put(final String rawPath,
 
         final ContentType contentType = ContentTypeLookup.findOrDefaultContentType(headers,
                 ContentType.APPLICATION_OCTET_STREAM);
+        Validate.notNull(contentType, "Content-Type header value must not be null");
```

Comment:
Why are we doing these checks when we weren't before?

Reply:
I saw it as a chance to just make sure that the code's more robust against errors and wanted to improve the ```[DEBUG]``` level.

Reply:
Let's leave the checks out because a user could pass null as the default content type to do whatever apache http client does by default. My intuition tells me that this check could lead to an unanticipated bug due to unexpected user behavior.

Reply:
Sure thing!